sdk/log/processor.go

```diff
@@ -35,6 +35,40 @@ type Processor interface {
 	// to create a copy that shares no state with the original.
 	OnEmit(ctx context.Context, record *Record) error
 
+	// Shutdown is called when the SDK shuts down. Any cleanup or release of
+	// resources held by the exporter should be done in this call.
+	//
+	// The deadline or cancellation of the passed context must be honored. An
+	// appropriate error should be returned in these situations.
+	//
+	// After Shutdown is called, calls to Export, Shutdown, or ForceFlush
+	// should perform no operation and return nil error.
+	Shutdown(ctx context.Context) error
+
+	// ForceFlush exports log records to the configured Exporter that have not yet
+	// been exported.
+	//
+	// The deadline or cancellation of the passed context must be honored. An
+	// appropriate error should be returned in these situations.
+	ForceFlush(ctx context.Context) error
+}
+
+// FilterProcessor is a Processor that knows, and can identify, what Record it
+// will process or drop when it passed to OnEmit.
+//
+// This is useful for users of logging APIs that want to know if they should
+// perform complex operations to emit a Record if it will be processed, or if
+// not if it will be dropped.
+//
+// This is an optional interface, and Processor implementations are not
+// required to support it. Implementations that choose to support this are
+// expected to re-evaluate the Records passed to OnEmit, it is not expected
+// that the caller to OnEmit will check if that Record will be processed or
+// dropped.
+//
+// This should only be implemented for Processors that can make reliable enough
+// determination of this prior to processing a Record.
+type FilterProcessor interface {
```

Comment:
`WithProcessor` docs should be updated and reference the processor can also implement `FilterProcessor`

Reply:
I've updated the `Processor` interface documentation to reference the ability to implement the `FilterProcessor`. I added the documentation here instead of `WithProcessor` as the interface is where implementer will be referencing and `WithProcessor` is something end-users will be referencing. The end-user does not need to be told about `FilterProcessor` when adding a `Processor`, it is the author of the `Processor` that need this information.